src/components/molecules/UserProfilePicture/UserProfilePicture.tsx

```diff
@@ -1,169 +1,115 @@
-const UserProfilePicture: React.FC<UserProfilePictureProp> = ({
-  isAudioEffectDisabled = true,
-  miniAvatars = false,
-  avatarClassName = "profile-icon",
-  avatarStyle,
-  containerStyle,
-  reactionPosition,
+export const UserProfilePicture: React.FC<UserProfilePictureProp> = ({
   user,
+  isAudioEffectDisabled = true,
+  containerClassName,
+  reactionPosition = "right",
+  // miniAvatars = false,
 }) => {
-  // @debt some of the redux patterns exist for this, but I don't believe anything actually uses them/calls this at the moment
-  const muteReactions = useSelector((state) => state.room.mute);
-  const { openUserProfileModal } = useProfileModalControls();
-
-  const [pictureUrl, setPictureUrl] = useState("");
-
-  const randomAvatarUrl = (id: string) =>
-    "/avatars/" +
-    RANDOM_AVATARS[Math.floor(id?.charCodeAt(0) % RANDOM_AVATARS.length)];
-
-  const avatarUrl = useCallback(
-    (id: string, anonMode?: boolean, pictureUrl?: string) => {
-      if (anonMode || !id) {
-        return setPictureUrl(DEFAULT_PROFILE_IMAGE);
-      }
-
-      if (!miniAvatars && pictureUrl) {
-        return setPictureUrl(pictureUrl);
-      }
-
-      if (miniAvatars) {
-        return setPictureUrl(randomAvatarUrl(id));
-      }
-
-      return setPictureUrl(DEFAULT_PROFILE_IMAGE);
-    },
-    [miniAvatars]
-  );
+  const userId = user?.id;
 
-  useEffect(() => {
-    avatarUrl(user.id, user.anonMode, user.pictureUrl);
-  }, [avatarUrl, user.anonMode, user.id, user.pictureUrl]);
-
-  const venueId = useVenueId();
-  const reactions = useReactions(venueId);
-
-  const typedReaction = reactions ?? [];
-
-  const messagesToBand = typedReaction.find(
-    (r) => r.reaction === "messageToTheBand" && r.created_by === user.id
-  ) as MessageToTheBandReaction | undefined;
+  const { openUserProfileModal } = useProfileModalControls();
 
-  const imageErrorHandler = useCallback(
-    (
-      event: HTMLImageElement | React.SyntheticEvent<HTMLImageElement, Event>
-    ) => {
-      const randomAvatar = randomAvatarUrl(user.id);
-      setPictureUrl(randomAvatar);
+  const openProfileModal = useCallback(() => openUserProfileModal(user), [
+    openUserProfileModal,
+    user,
+  ]);
 
-      (event as HTMLImageElement).onerror = null;
-      (event as HTMLImageElement).src = randomAvatar;
-    },
-    [user.id]
+  // const pictureUrl = getAvatarUrl({ user, miniAvatars });
+
+  // @debt useImage tries to load the images twice, which is made worse by us not caching images retrieved from firebase,
+  //  it's only used to handle the edgecase of showing a default when images are missing. Can we live without it?
+  // const { loadedImageUrl: pictureUrl } = useImage({
+  //   src: avatarUrl({ user, miniAvatars }),
+  //   fallbackSrc: () => randomAvatarUrl(user.id),
+  // });
+
+  // const containerVars = useCss({
+  //   "--user-profile-picture-avatar-url": pictureUrl
+  //     ? `url(${pictureUrl})`
+  //     : `url(${DEFAULT_PROFILE_IMAGE})`,
+  // });
```

Comment:
Deleted/comment?

Reply:
Added `@debt` comments to explain the commented code blocks in https://github.com/sparkletown/sparkle/pull/1323/commits/5f2acbd92b4f96c175bf39ab1ffc0c0bb7d92624